### PR TITLE
feat: Add API to get sample apps related to a template

### LINKED ISSUE #14786: [Task] GET Apps connected to a template. (BE)
The Get API for Backend to get connected apps of a template.

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
This PR adds query params to the get similar templates API.

Fixes #14786

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?
- Manually

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java [hunk 1]
```diff
@@ -55,10 +55,16 @@ public ApplicationTemplateServiceCEImpl(CloudServicesConfig cloudServicesConfig,
...
     @Override
-    public Flux<ApplicationTemplate> getSimilarTemplates(String templateId) {
-        final String apiUrl = String.format("%s/api/v1/app-templates/%s/similar?version=%s",
-                cloudServicesConfig.getBaseUrl(), templateId, releaseNotesService.getReleasedVersion()
-        );
+    public Flux<ApplicationTemplate> getSimilarTemplates(String templateId, MultiValueMap<String, String> params) {
+        UriComponents uriComponents = UriComponentsBuilder
+                .fromUriString(cloudServicesConfig.getBaseUrl())
+                .pathSegment("api/v1/app-templates", templateId, "similar")
+                .queryParams(params)
```
Comment (sharat87) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java:62]: Looks like the params is passed on as received without looking inside? Can we validate this map? Or, even better, just take the params we want/expect from this map, and pass _only_ those that we know.

**Review by sharat87 — CHANGES_REQUESTED** [review]
Just one change I'd like to see, otherwise good to go.

File: app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java (hunk 1 above)
Comment (nayan-rafiq) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java:62]: @sharat87 This API is actually a proxy to the underlying cloud services API, so I didn't validate the map attributes here as it's validated in the cloud services API. If in future we need an additional parameter, we only need to change in the cloud services. Do you think validating the parameters here as well is necessary?

Comment (sharat87) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/services/ce/ApplicationTemplateServiceCEImpl.java:63]: Yeah, figured, but it feels a little implicit. It's probably just me being paranoid again, 🤷

**Review by sharat87 — APPROVED** [review]

**Pull request merged**